### ISSUE TITLE
Popup `graph_menu` on pin drop

### ISSUE THREAD

**Issue by kang-sw** [zakarumych/egui-snarl#10]
Currently right clicking on the panel is the only way to add graph menu. It'd be nice to making graph menu open on dragged pin is dropped to empty space of panel. Since we can track from which node what pin is being dragged, it'd be possible to auto-connect newly created node from the menu with the node pin currently being dragged.

Pseudo code:
```
let pin = await viewer.drag
await pin.dropped

let new_node = await viewer.graph_menu(..)

// Iterate until we find matching pin

if pin.is_input {
    for output_pin in new_node.outputs {
        
        if viewer.try_connect(pin, output_pin) {
            return;
        }
    }
} else {
    for input_pin in new_node.inputs {
        if viewer.try_connect(input_pin, pin) {
            return;
        }
    }
}

// if no pin is matching, it's just okay ...
```

**Comment by zakarumych** [comment]
Yes, it'd be very useful. It may use same `SnarlViewer::graph_menu` or a separate method to populate that menu.
Would you like to work on this?

**Comment by kang-sw** [comment]
Ah, yes. I'll try. What is your opinion for adding new trait method? I think it'd be useful to know which node is being connected.

**Comment by kang-sw** [comment]
And for this, it seems 'connect()' method does not return any result when invoked. Is it okay to change the signature to return `Result` or boolean? It will be breaking change.

And if it's okay to return `Result`, what should the error type be?

[link](https://docs.rs/egui-snarl/latest/egui_snarl/ui/trait.SnarlViewer.html#method.connect)
```rust
    /// Asks the viewer to connect two pins.
    ///
    /// This is usually happens when user drags a wire from one node's output pin to another node's input pin or vice versa.
    /// By default this method connects the pins and returns `Ok(())`.
    #[inline]
    fn connect(&mut self, from: &OutPin, to: &InPin, snarl: &mut Snarl<T>) {
        snarl.connect(from.id, to.id);
    }
```

**Comment by zakarumych** [comment]
`SnarlViewer::connect` is command to add connection.
Currently there's nothing to do if it fails, or intermediate note is added, or all nodes just vanish in result :) 

Result in return can be added if there will be something to do about it.

UI commands should be self-contained. So one UI action should not cause multiple commands.
